### ISSUE TITLE
tf_dataset.map(sequential_model(x)) results in Cannot convert a symbolic tf.Tensor (args_0:0) to a numpy array.

### ISSUE THREAD

**Issue by divyashreepathihalli** [keras-team/keras#18396]
Error with JAX and Torch backend

when dataset = tf_data.Dataset.from_tensor_slices((images, labels)) is used with map function to call a sequential model of augmentation layers results in Cannot convert a symbolic tf.Tensor (args_0:0) to a numpy array. error
[repro here](https://colab.sandbox.google.com/drive/1TFZv8ahr3a1URSFrRObeWaVdSv1DYp6B#scrollTo=Vxx3Wz9sVkKz)

**Comment by fchollet** [comment]
Yes, that's expected. `Sequential` cannot be used with tf.data unless you're running on the TF backend.
